Source/Proxy/IPublicCallContext.cs

```diff
@@ -0,0 +1,21 @@
+using System.Reflection;
+
+namespace Moq.Proxy
+{
+	/// <summary>
+	/// Class that represents the method call currently being executed. 
+	/// Allows for custom interception for open generic methods
+	/// </summary>
+	public interface IPublicCallContext
+	{
+		/// <summary>
+		/// Arguments passed into the method
+		/// </summary>
+		object[] Arguments { get; }
+
+		/// <summary>
+		/// Closed generic method currently being executed.
+		/// </summary>
+		MethodInfo Method { get; }
+	}
+}
```

Comment:
I do not yet quite understand why you chose to split `IPublicCallContext` off `ICallContext` and make it public? Why is this required at all? As far as I can see, it's only used for the new overload of `.Returns()`. To me, this seems like an orthogonal (i.e. completely separate) feature. Could this new public contract possibly be moved off into a separate PR? Or will the new `It.AnyType` feature not work without this?

Additionally, if this has to remain public in order for `It.AnyType` to work, I think it *might* be better to move it into the main `Moq` namespace. Not entirely certain on that point, though.